derp/derp_server.go

```diff
@@ -445,6 +450,22 @@ func (s *Server) initMetacert() {
 	if err != nil {
 		log.Fatal(err)
 	}
+
+	type basicConstraints struct {
+		IsCA       bool `asn1:"optional"`
+		MaxPathLen int  `asn1:"optional"`
+	}
+	extEnc, err := asn1.Marshal(basicConstraints{IsCA: true, MaxPathLen: -1})
+	if err != nil {
+		log.Fatal(err)
+	}
```

Comment:
Add a comment linking to https://pkg.go.dev/crypto/x509#Certificate.MaxPathLen.

Reply:
Actually simplified this a ton after reading those docs!